pkg/analyze/symptoms/rules/storage.go

```diff
@@ -0,0 +1,290 @@
+			}))
+
+	return symptoms.NewSymptomTreeLeaf("StorageClass missing", notDeployedStorageClass)
+}
+
+func buildNodeConfigSymptoms() symptoms.SymptomTreeNode {
+	// Scenario #4: E2Es for misconfigured NodeConfigs
+	nodeConfigClusterWideNonexistentVolume := symptoms.NewSymptom(
+		"NodeConfig doesn't provision storage for local-csi-driver's volumes-dir",
+		[]string{
+			"driver is unable to provide storage due to NodeConfig being misconfigured",
+			"this may be a false-positive in clusters which don't use NodeConfig to mount `volumes-dir` of local-csi-drivers' Pods",
+		},
+		[]string{
+			"fix the NodeConfig (see node-config's #TODO: node config path# error conditions), then rolling restart local-csi-driver",
+		},
+		selector.
+			New().
+			Select("scylla-cluster", selector.Type[*scyllav1.ScyllaCluster](), func(c *scyllav1.ScyllaCluster) (bool, error) {
+				return !symptoms.MeetsCondition(c, "Available", "True"), nil
+			}).
+			Select("scylla-pod", selector.Type[*v1.Pod](), func(p *v1.Pod) (bool, error) {
+				return p.Status.Phase != v1.PodSucceeded, nil
+			}).
+			Select("pod-pvc", selector.Type[*v1.PersistentVolumeClaim](), func(pvc *v1.PersistentVolumeClaim) (bool, error) {
+				return pvc.Status.Phase == v1.ClaimPending && pvc.Spec.StorageClassName != nil, nil
+			}).
+			// TODO: check every storage class
+			Select("storage-class", selector.Type[*storagev1.StorageClass](), nil).
+			Select("csi-driver", selector.Type[*storagev1.CSIDriver](), nil).
+			Select("scylla-node", selector.Type[*v1.Node](), nil).
+			Select("node-config", selector.Type[*scyllav1alpha1.NodeConfig](), nil).
+			Select("csi-pod", selector.Type[*v1.Pod](), func(d *v1.Pod) (bool, error) {
+				for _, cont := range d.Spec.Containers {
+					if cont.Name == csiDriverContainerName {
+						return true, nil
+					}
+				}
+				return false, nil
+			}).
+			Relate("scylla-cluster", "scylla-pod", func(c *scyllav1.ScyllaCluster, p *v1.Pod) (bool, error) {
+				return c.Name == p.Labels["scylla/cluster"], nil
+			}).
+			Relate("scylla-pod", "pod-pvc", func(p *v1.Pod, pvc *v1.PersistentVolumeClaim) (bool, error) {
+				for _, volume := range p.Spec.Volumes {
+					if volume.PersistentVolumeClaim != nil {
+						volumePvc := *volume.PersistentVolumeClaim
+						if volumePvc.ClaimName == pvc.Name {
+							return true, nil
+						}
+					}
+				}
+				return false, nil
+			}).
+			Relate("pod-pvc", "storage-class", func(pvc *v1.PersistentVolumeClaim, s *storagev1.StorageClass) (bool, error) {
+				return *pvc.Spec.StorageClassName == s.Name, nil
+			}).
+			Relate("storage-class", "csi-driver", func(s *storagev1.StorageClass, csi *storagev1.CSIDriver) (bool, error) {
+				return s.Provisioner == csi.Name, nil
+			}).
+			Relate("scylla-node", "scylla-cluster", func(n *v1.Node, c *scyllav1.ScyllaCluster) (bool, error) {
+				// TODO: relate the node to the cluster (with node placement rules?)
```

Comment:
ditto

Reply:
Here, AFAIU, the problem is that there's no node meeting pod requirements, so the pod can't be scheduled. I needed to somehow check if a node is part of a cluster, but found no feasible way to do this, except from node placement rules. I'm not sure if there's some placement rules matcher, and writing one from scratch may not be worth it